<!-- SLAM Gmapping package configuration parameters -->

<launch>
    <include file="$(find realsense2_camera)/launch/rs_rgbd.launch"/>

<!-- run sweep_node node -->
    <node name="sweep_node" pkg="sweep_ros" type="sweep_node" output="screen">
        <param name="serial_port" type="string" value="/dev/ttyUSB1"/>
        <param name="serial_baudrate" type="int" value="115200"/>
        <param name="frame_id" type="string" value="laser_frame"/>
        <param name="rotation_speed" type="int" value="2"/>
        <param name="sample_rate" type="int" value="1000"/>
    </node>

    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">

        <remap from="cloud_in" to="pc2"/>
        <rosparam>
            target_frame: laser_frame # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.001
            min_height: -1.0
            max_height: 1.0

            angle_min: -3.14 # -M_PI/2
            angle_max: 3.14 # M_PI/2
            angle_increment: 0.01 # M_PI/360.0
            scan_time: 0.1
            range_min: 0.25
            range_max: 40.0
            use_inf: true

            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>

    </node>

</launch>
 
